src/Features/CSharp/Portable/RemoveUnnecessarySuppression/CSharpRemoveUnnecessarySuppressionCodeFixProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppression
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal partial class CSharpRemoveUnnecessarySuppressionCodeFixProvider : CodeFixProvider
```

Comment:
Any reason this derives from `CodeFixProvider` and uses a custom `FixAllProvider` that derives from `DocumentBasedFixAllProvider` instead of this type deriving from `SyntaxEditorBasedCodeFixProvider`?

Reply:
yes.  SyntaxEditorBasedCodeFixProvider doesn't work well when you have *multiple* different code actions to provide, and you want each one to have a fixall that behaves unique to it.

Reply:
this is the pattern that works when you need to pass the equivalence key along to know what behavior the fix-all should be following.  in this case, you can 'fix all, while preserving semantics' or 'fix all, while inverting semantics'.